dartagnan/src/main/java/com/dat3m/dartagnan/wmm/Wmm.java

```diff
@@ -104,85 +110,127 @@ public void initialise(VerificationTask task, SolverContext ctx) {
         }
     }
 
-    /*
-        This method computes all the min sets, max sets and possibly encode sets
-        of axiom-relevant relations.
-     */
-    public void performRelationalAnalysis(boolean computeEncodeSets) {
-        if (this.task == null) {
-            throw new IllegalStateException("The WMM needs to get initialised first.");
-        }
-
-        for (RecursiveGroup recursiveGroup : recursiveGroups) {
-            recursiveGroup.initMaxTupleSets();
-            recursiveGroup.initMinTupleSets();
-        }
-
-        for (Axiom ax : axioms) {
-            ax.getRelation().getMaxTupleSet();
-        }
-
-        for(String relName : baseRelations){
-            relationRepository.getRelation(relName).getMaxTupleSet();
-        }
-
-        if (computeEncodeSets) {
-            for (Axiom ax : axioms) {
-                ax.getRelation().addEncodeTupleSet(ax.getEncodeTupleSet());
-            }
-
-            Collections.reverse(recursiveGroups);
-            for (RecursiveGroup recursiveGroup : recursiveGroups) {
-                recursiveGroup.updateEncodeTupleSets();
-            }
-        }
-    }
-
-    // This methods initializes all relations and encodes all base relations
-    // and recursive groups (why recursive groups?)
-    // It also triggers the computation of may and active sets!
-    // It does NOT encode the axioms nor any non-base relation yet!
-    public BooleanFormula encodeBase(SolverContext ctx) {
-        performRelationalAnalysis(true);
-        BooleanFormulaManager bmgr = ctx.getFormulaManager().getBooleanFormulaManager();
+	/**
+	 * Computes the may-sets, must-sets and possibly also must-not and active sets.
+	 * @param computeMustNotSets Whether must-not sets get computed.
+	 * @param computedActiveSets Whether active sets get computed.
+	 */
+	public void performRelationAnalysis(boolean computeMustNotSets, boolean computedActiveSets) {
```

Comment:
The old code used to check that the task was not null ... why was the check removed?

Reply:
The method was added in 'development', merged into this branch, inlined into `encodeRelations` again, then outlined yet again without the check.

Reply:
done as told